### PR TITLE
Merge `package:args`

### PR DESCRIPTION
**Pull request by mosuem**
- [x] Move and fix workflow files and badges in the README.md
- [x] Update pubspec.yaml to point to the new repository
- [x] Rev the version of the package, so that pub.dev points to the correct site
- [x] Add the package to the top-level readme of the monorepo
- [ ] Merge the PR with 'Create a merge commit', enable `Allow merge commits` if not yet enabled on this repository
- [ ] Update the auto-publishing settings on https://pub.dev/packages/args/admin
- [ ] Add the following text to https://github.com/dart-lang/args/:'

```
> [!IMPORTANT]  
> This repo has moved to https://github.com/dart-lang/core/tree/main/pkgs/args
```

- [ ] Publish using the autopublish workflow
- [ ] Push tags to GitHub using `git tag --list 'args*' | xargs git push origin`
- [ ] Close open PRs in dart-lang/args with the following message:

`Closing as the [dart-lang/args](https://github.com/dart-lang/args) repository is merged into the [dart-lang/core](https://github.com/dart-lang/core) monorepo. Please re-open this PR there!`
      
- [ ] Transfer issues by running `dart run pkgs/repo_manage/bin/report.dart transfer-issues --source-repo dart-lang/args --target-repo dart-lang/core --add-label package:args --apply-changes`
- [ ] Archive https://github.com/dart-lang/args/

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: .github/labeler.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Configuration for .github/workflows/pull_request_label.yml.
```
Comment (devoncarew) [inline .github/labeler.yaml:1]: Not for this PR, but we'll also need a `workflows/pull_request_labeller.yml` file for this config to take effect.

File: .github/workflows/args.yaml [hunk 2]
```diff
@@ -4,13 +4,23 @@ on:
...
     branches: [ main ]
+    paths:
+      - '.github/workflows/args.yaml'
+      - 'pkgs/args/**'
   schedule:
     - cron: "0 0 * * 0"
 env:
   PUB_ENVIRONMENT: bot.github
 
+defaults:
```
Comment (devoncarew) [inline .github/workflows/args.yaml:20]: Ah, this is nice; I've been setting the cwd for each job config. We should definitely prefer to do as a workflow file default.

**Review by devoncarew — APPROVED** [review]
This looks good to me.

Note that there is an active PR that's just about to land; https://github.com/dart-lang/args/pull/287. We should wait for that before closing open PRs / switching the repo-of-record.

File: .github/labeler.yaml (hunk 1 above)
Comment (devoncarew) [reply .github/labeler.yaml:1]: https://github.com/dart-lang/labs/blob/main/.github/workflows/pull_request_label.yml or similar.

**Comment by devoncarew** [comment]
I created a PR against the open (active) PR in args; we may be delayed a bit as I believe Sigurd is OOO.

https://github.com/sigurdm/args/pull/1

Perhaps we should just land his PR and address the lints after the fact? Or create a duplicate PR and fix+land that.

**Comment by mosuem** [comment]
> create a duplicate PR and fix+land that

+1, the PR is already approved and not too big, so should be easy enough to recreate.

File: .github/workflows/args.yaml (hunk 2 above)
Comment (mosuem) [reply .github/workflows/args.yaml:20]: Same, just tried this and it seems to work.

**Pull request merged**